### PR TITLE
Back to top button

### LINKED ISSUE #3763: Add a "back to top" tooltip
The new burotel requires a lot of scrolling across rooms/desks. People who are not that mouse-friendly may have a bad time rolling their mousewheels for Kilometers at at time.
I am thinking of a discrete tooltip in the bottom right corner, which when clicked scrolls back up. But suggestions are welcome.
This could be interesting for the core system.

### PR DESCRIPTION
**Pull request by mvidalgarcia**
closes #3763 

Related to: https://github.com/indico/indico-plugins-cern/pull/43

### CODE REVIEW

File: indico/web/client/js/react/components/ScrollButton.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+export default function ScrollButton({visible}) {
+    function scrollToTop() {
+        window.scroll({left: 0, top: 0, behavior: 'smooth'});
+    }
+
+    return (
+        <Popup trigger={<Button icon="angle up"
+                                onClick={scrollToTop}
+                                styleName="scroll-btn"
+                                className={visible ? 'visible' : ''} />}
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/ScrollButton.jsx]: It looks like you can use `styleName` for this and keep the conditional inside `styleName`, and thus remove the `:global()` in the CSS:

https://github.com/gajus/babel-plugin-react-css-modules#runtime-stylename-resolution

Comment (mvidalgarcia) [reply indico/web/client/js/react/components/ScrollButton.jsx]: I wanted to avoid the repetition of the `scroll-btn` class

**Comment by ThiefMaster** [comment]
scrollBtnVisible -> scrollButtonVisible? or does this result in too-long lines somewhere?

File: indico/web/client/js/react/components/StickyWithScrollBack.module.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+@import 'rb_new:styles/palette';
+
+.sticky-filters {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/StickyWithScrollBack.module.scss]: is "filters" really a good name here now that it's outside the RB module?

File: indico/web/client/js/react/components/ScrollButton.module.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+@import 'rb_new:styles/palette';
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/ScrollButton.module.scss]: global components shouldn't import stuff from rb

File: webpack/base.js [hunk 4]
```diff
@@ -108,21 +108,27 @@ export function webpackDefaults(env, config, bundles) {
     }
 
     /**
-     * This function resolves SASS files using a module:path syntax
+     * This function resolves SASS files using a [~][module:]path syntax
      */
     function sassResolver(url) {
-        const match = url.match(/^(~?\w+):([\w/-_]+)/);
+        const match = url.match(/^(~?)((\w+):)?([\w/-_]+)/);
```
Comment (ThiefMaster) [inline webpack/base.js]: use `(?:...)` for a non-matching group (currently you don't use `[1]`)

**Pull request merged**